R/DataPulls.R

```diff
@@ -591,4 +598,170 @@ getCohortMethodResult <- function(connection = NULL,
+#' @return                     Tibble. 
+#'
+#' @examples
+#' \dontrun{
+#' connectionDetails <- createConnectionDetails(dbms = "postgresql",
+#'                                            user = "joe",
+#'                                            password = "secret",
+#'                                            server = "myserver")
+#' example1 <- getPower(connection = connectionDetails,
+#'                      databaseSchema = 'database.schema',
+#'                      targetId = 100,
+#'                      comparatorId = 200,
+#'                      outcomeId = 111,
+#'                      databaseId = 'HCUP data',
+#'                      analysisId = 2)
+#' example2 <- getPower(cohortMethodResult = cohortMethodResult,
+#'                      cohortMethodAnalysis = cohortMethodAnalysis,
+#'                      targetId = 100,
+#'                      comparatorId = 200,
+#'                      outcomeId = 111,
+#'                      databaseId = 'HCUP data',
+#'                      analysisId = 2)
+#' }
+#' @export
+#'
+getPower <- function(connection = NULL,
+                     databaseSchema = NULL,
+                     cohortMethodResult = NULL,
+                     cohortMethodAnalysis = NULL,
+                     targetId,
+                     comparatorId,
+                     outcomeId,
+                     databaseId,
+                     analysisId) {
+  errorMessage <- checkmate::makeAssertCollection()
+  checkmate::assertInt(targetId, add = errorMessage)
+  checkmate::assertInt(comparatorId, add = errorMessage)
+  checkmate::assertInt(outcomeId, add = errorMessage)
+  checkmate::assertInt(analysisId, add = errorMessage)
+  checkmate::assertScalar(databaseId, add = errorMessage)
+  checkmate::assertCharacter(databaseId, add = errorMessage)
+  
+  if (!is.null(connection)) {
+    checkmate::assertCharacter(databaseSchema)
+    checkmate::reportAssertions(errorMessage)
+    
+  } else if (is.null(connection)) {
+    checkmate::assertDataFrame(cohortMethodResult, add = errorMessage)
+    checkmate::assertDataFrame(cohortMethodAnalysis, add = errorMessage)
+    checkmate::reportAssertions(errorMessage)
+  } else {
+    stop(
+      "No connection for an RDMS provided.
+         No R-data frame object cohortMethodResult and cohortMethodAnalysis provided"
+    )
+  }
+  
+  cohortMethodResult <-
+    .getcohortMethodResult(
+      connection = connection,
+      databaseSchema = databaseSchema,
+      cohortMethodResult = cohortMethodResult,
+      targetId = targetId,
+      comparatorId = comparatorId,
+      outcomeId = outcomeId,
+      databaseId = databaseId,
+      analysisId = analysisId
+    )
+  
+  cohortMethodAnalysis <-
+    .getcohortMethodAnalysis(
+      connection = connection,
+      databaseSchema = databaseSchema,
+      cohortMethodAnalysis = cohortMethodAnalysis,
+      analysisId = analysisId
+    ) %>%
+    dplyr::select(-definition)
+  
+  #fixed parameters
+  alpha <- 0.05
```

Comment:
Why are these parameters not exposed as function arguments?

Reply:
Agree - lets parameterize